documentation/docs/examples/timeline.mdx

```diff
@@ -0,0 +1,213 @@
+
+```js runnable
+// import { NodeCollectionBase, DefaultNodeAppearance, PropertyFilterNodeCollection } from '@cognite/reveal';
+// import { TimelineTool, Keyframe } from '@cognite/reveal/tools';
+
+function createProgressBarContainer() {
+  const container = document.createElement('div');
+  //container.innerText = `Progress`;
+  container.style.cssText = `
+    background: #333;
+    border-radius: 13px;
+    width: 500px;
+    height: 25px;
+    padding: 3px;
+    align-items: center;
+    justify-content: center;
+    position: relative;
+    transform: translate(-50%, -50%);
+    left: 50%;
+    bottom: 20px;
+  `;
+  return container;
+}
+
+function createProgressBar() {
+  const progress = document.createElement('div');
+  progress.style.cssText = `
+    /* Progress after moving to next Keyframe */
+    content: '';
+    display: block;
+    background: DodgerBlue;
+    width: 0%;
+    height: 20px;
+    border-radius: 9px;
+    position: relative;
+  `;
+  return progress;
+}
+
+model.setDefaultNodeAppearance(DefaultNodeAppearance.Ghosted);
+
+const timeline = new TimelineTool(model);
+
+// div element Cognite3DViewer to insert into the canvas
+const domElement = document.getElementById('demo-wrapper');
+const container = createProgressBarContainer();
+const progressBar = createProgressBar();
+container.appendChild(progressBar);
+domElement.appendChild(container);
+
+const nodes = new AssetNodeCollection(sdk, model);
+const keyframeAssetNode = timeline.createKeyframe(new Date('2021-10-05'));
+keyframeAssetNode.assignStyledNodeCollection(nodes, { renderGhosted: false, outlineColor: NodeOutlineColor.Cyan });
+
+nodes.executeFilter({ assetId: 4222791488928479 });
+const keyframeAssetID = timeline.createKeyframe(new Date('2021-10-06'));
+keyframeAssetID.assignStyledNodeCollection(nodes, { renderGhosted: false, outlineColor: NodeOutlineColor.Orange });
+
+const allPipes = new PropertyFilterNodeCollection(sdk, model);
+allPipes.executeFilter({'PDMS': {'Type': 'PIPE'}});
+const keyframeAllPipes = timeline.createKeyframe(new Date('2021-10-07'));
+keyframeAllPipes.assignStyledNodeCollection(allPipes, { renderGhosted: false, color: [40, 200, 20] });
+
+const stuMechNodes = new PropertyFilterNodeCollection(sdk, model);
+stuMechNodes.executeFilter({'PDMS': {':capDisc': 'MECH', 'Purpose': 'STU' }});
+const keyframeMechNodes = timeline.createKeyframe(new Date('2021-10-08'));
+keyframeMechNodes.assignStyledNodeCollection(stuMechNodes, { renderGhosted: false, color: [80, 100, 120] });
+
+const playLength = 25000;
+
+timeline.play(new Date('2021-10-05'), new Date('2021-10-09'), playLength);
+
+let refDate = 0;
+
+timeline.subscribe('dateChanged', (data) => {
+  if(refDate === 0) {
+    refDate = data.date.getTime();
+  }
+  const width = ((data.date.getTime() - refDate) / playLength) * 100;
+  const widthPercentage = width.toString() + '%';
+  progressBar.style.width = widthPercentage;
+})
```

Comment:
```suggestion
const durationMs = 25000;
const startDate = new Date('2021-10-05');
const endDate = new Date('2021-10-09');
timeline.play(startDate, endDate, durationMs);

timeline.subscribe('dateChanged', (data) => {
  const progress = ((data.date.getTime() - startDate.getTime()) / (endDate.getTime() - startDate().getTime()) * 100;
  progressBar.style.width = `${progress}%`;
})
```

The `refDate` was a bit confusing. Also, why divide by `playLength` in computation of width?

Must be tested.

Reply:
The suggested changes without using reference date and `durationMs` / `playLength` will not work for `${progress}%`. Both values are need to plot the progression

Reply:
I don't understand why - there should not be a need to include playLength in this computation. The progress is purely a result of start + end date and "virtual" date?

Reply:
If the user provides a wrong start date and end date, than progress value will not be right even if we handle the wrong start date & end date in the tool

Reply:
The progress should be related to user input - if user provides start=2020-01-01 and end=2020-07-01, the progress should be ~50% when the virtual date is 2020-04-01